api4/plugin.go

```diff
@@ -100,22 +100,84 @@ func installPluginFromUrl(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	downloadUrl := r.URL.Query().Get("plugin_download_url")
+	pluginDownloadUrl := r.URL.Query().Get(model.PLUGIN_URL_STRING)
+	signatureDownloadUrl := r.URL.Query().Get(model.SIGNATURE_URL_STRING)
+	downloadUrls := map[string]string{
+		model.PLUGIN_URL_STRING: pluginDownloadUrl,
+	}
+	if signatureDownloadUrl != "" {
+		downloadUrls[model.SIGNATURE_URL_STRING] = signatureDownloadUrl
+	}
+
+	downloadedFiles := downloadFiles(c, downloadUrls)
```

Comment:
rather than passing the context, would it be cleaner if `downloadFiles` returned an error instead? more go-idiomatic to check for error than checking for a key to check for an error.